lighthouse-core/lib/network-request.js

```diff
@@ -453,6 +453,24 @@ class NetworkRequest {
     return requestId.replace(/(:redirect)+$/, '');
   }
 
+  /**
+   * Return the body of the response with the given ID. Rejects if getting the
+   * body times out.
+   * @param {LH.Gatherer.FRProtocolSession} session
+   * @param {string} requestId
+   * @param {number} [timeout]
+   * @return {Promise<string>}
+   */
+  static async getRequestContent(session, requestId, timeout = 1000) {
```

Comment:
This felt like the most convenient location for this function. I'm open to other suggestions though.

Reply:
This was my first thought as well, but it does feel weird to introduce `session` into this file. My other thoughts were on NetworkMonitor?

Reply:
It also feels weird to create a `new NetworkMonitor(session)` just to use this function with `session`. I think I prefer leaving as is.

Reply:
if it's a gathering helper method it seems like it should live under `driver/`, but it seems like it should just go in a new file if none of the existing helper modules are a good home for it? If the stricter timeout is important so we want people getting response content this way, it should preferably not be tucked in somewhere that's already relatively large and has an important separate purpose or people are just going to call `Network.getResponseBody` directly because they won't know it's there

Reply:
actually, looking at the original PR (https://github.com/GoogleChrome/lighthouse/pull/4718), the timeout was specifically for working around using the protocol via the extension `chrome.debugger` API and there wasn't an issue in devtools/CLI. The extension doesn't use the protocol anymore, so should we just get rid of the method and everyone can call `sendCommand('Network.getResponseBody', {requestId})` directly?

Reply:
> it should preferably not be tucked in somewhere that's already relatively large and has an important separate purpose or people are just going to call Network.getResponseBody directly because they won't know it's there

This was my concern with mingling session work in this file as well. You've boosted me beyond "could go either way" to "should probably stay in driver/ somewhere" :)

> so should we just get rid of the getRequestContent method and everyone can call sendCommand('Network.getResponseBody', {requestId}) directly?

I would lean no. Defaulting this to a low timeout is a good thing, and the detail with `NetworkRequest.getRequestIdForBackend` is easy to forget, so bundling those together at least is necessary (otherwise your exact same argument of missing an argument tucked away somewhere is likely to happen here).

Reply:
ah yeah, I missed `NetworkRequest.getRequestIdForBackend` in there.

The timeout still seems like cruft (should the comment point to this discussion now since #4718 can't happen anymore? :P) but it's pre-existing so 🤷

Reply:
> It also feels weird to create a new NetworkMonitor(session) just to use this function with session. I think I prefer leaving as is.

I was thinking just a static method like it is here. The alternative I see is a solo file.

Reply:
I have moved it to a new file, `network.js`